File: chronos/src/Chronos.ProfilerAgent/ByteMarshaler.cpp
#include "StdAfx.h"
#include "ByteMarshaler.h"

void CByteMarshaler::Marshal(__byte value, CBaseStream* stream)
{
	stream->Write(&value, TypeSize::_BYTE);
}

__byte CByteMarshaler::Demarshal(CBaseStream* stream)
{
	__byte value = 0;
	stream->Read(&value, TypeSize::_BYTE);
	return value;
}

File: Chronos/source/Chronos/Marshaling/Generic/ByteMarshaler.cs
using System;
using System.IO;

namespace Chronos.Marshaling
{
    public class ByteMarshaler : GenericMarshaler<byte>
    {
        protected override void MarshalInternal(byte value, Stream stream)
        {
            Marshal(value, stream);
        }

        protected override byte DemarshalInternal(Stream stream)
        {
            return Demarshal(stream);
        }

        public static void Marshal(byte value, Stream stream)
        {
            stream.WriteByte(value);
        }

        public static byte Demarshal(Stream stream)
        {
            return (byte) stream.ReadByte();
        }
    }
}
